Snowflake.Data/Client/SnowflakeDbDataReader.cs

```diff
@@ -255,6 +257,75 @@ public override int GetValues(object[] values)
             return count;
         }
 
+        public T GetObject<T>(int ordinal)
+            where T : class, new()
+        {
+            try
+            {
+                var rowType = resultSet.sfResultSetMetaData.rowTypes[ordinal];
+                var fields = rowType.fields;
+                if (fields == null || fields.Count == 0)
+                {
+                    throw new Exception("Method can be used only for structured object");

```

Comment:
Shouldn't we wrap it with SnowflakeDbException?

Reply:
fixed